File: rsdroid/src/main/java/net/ankiweb/rsdroid/BackendV1.java
package net.ankiweb.rsdroid;

import net.ankiweb.rsdroid.database.SQLHandler;

import java.io.Closeable;

import BackendProto.Backend;

public interface BackendV1 extends SQLHandler, net.ankiweb.rsdroid.RustBackend, net.ankiweb.rsdroid.Adbackend, Closeable {
    void openAnkiDroidCollection(Backend.OpenCollectionIn args) throws BackendException;

    boolean isOpen();

    /**
     * Downgrades the collection from Schema 16 to Schema 11
     * @param collectionPath The fully qualified path to collection.anki2
     * @throws BackendException The collection is not schema 16
     * @throws BackendException Collection is already open
     */
    void downgradeBackend(String collectionPath) throws BackendException;
}

File: rsdroid/src/main/java/net/ankiweb/rsdroid/BackendV1.kt
package net.ankiweb.rsdroid

import net.ankiweb.rsdroid.database.SQLHandler
import java.io.Closeable

interface BackendV1 : SQLHandler, Closeable {
    fun openCollection(collectionPath: String, mediaFolderPath: String, mediaDbPath: String, logPath: String, forceSchema11: Boolean);
    fun isOpen(): Boolean
}
